sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -105,7 +105,13 @@ case class CreateViewCommand(
     }
     val sessionState = sparkSession.sessionState
 
-    if (isTemporary) {
+    // 1) CREATE VIEW: create a temp view when users explicitly specify the keyword TEMPORARY;
+    //                 otherwise, create a permanent view no matter whether the temporary view
+    //                 with the same name exists or not.
+    // 2) ALTER VIEW: alter the temporary view if the temp view exists; otherwise, try to alter
```

Comment:
```SQL
      CREATE (OR REPLACE)? TEMPORARY? VIEW (IF NOT EXISTS)? tableIdentifier
        identifierCommentList? (COMMENT STRING)?
        (PARTITIONED ON identifierList)?
        (TBLPROPERTIES tablePropertyList)? AS query
```

Each flag corresponds to a keyword in the `CREATE VIEW DDL` command. 

- `OR REPLACE` -> `replace`
- `TEMPORARY ` -> `isTemporary `
- `IF NOT EXISTS` -> `allowExisting`

If we want to use the same command `CreateViewCommand` to process both `CREATE VIEW` and `ALTER VIEW`, it sounds reasonable to add a new flag. So far, I have not found a way to combine them.